src/main/java/Duke.java

```diff
@@ -26,95 +22,20 @@ public static void main(String[] args) throws DukeException {
-                    if (input.substring(0, 4).equals("todo")) {
-                        if (input.length() == 4) {
-                            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
-                        }
-                        tasks.addTask(new ToDo(input.substring(5)));
-                    } else if (input.substring(0, 5).equals("event")) {
-                        if (input.length() == 5) {
-                            throw new DukeException("☹ OOPS!!! The description of an event cannot be empty.");
-                        }
-                        int i = input.indexOf('/');
-                        String dateAndTime = input.substring(i + 4);
-                        DateTimeFormatter format = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
-                        LocalDateTime dateTime = LocalDateTime.parse(dateAndTime, format);
-                        tasks.addTask(new Event(input.substring(6, i - 1), dateTime));
-                    } else if (input.substring(0, 8).equals("deadline")) {
-                        if (input.length() == 8) {
-                            throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
-                        }
-                        int i = input.indexOf('/');
-                        String dateAndTime = input.substring(i + 4);
-                        DateTimeFormatter format = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
-                        LocalDateTime dateTime = LocalDateTime.parse(dateAndTime, format);
-                        tasks.addTask(new Deadline(input.substring(9, i - 1), dateTime));
-                    } else {
-                        throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
-                    }
-
-                    try {
-                        storage.addToFile(tasks.getTask(tasks.getNumOfTasks() - 1).toSaveString());
-                    } catch (IOException ex) {
-                        System.out.println("Cannot save new task in file");
-                    }
-
-                    System.out.println("Got it. I've added this task:");
-                    System.out.println("  " + tasks.getTask(tasks.getNumOfTasks() - 1));
-                    System.out.println("Now you have " + tasks.getNumOfTasks() +
-                            (tasks.getNumOfTasks() == 1 ? " task" : " tasks") + " in the list.");
-                }
-            } catch (DukeException ex) {
-                System.out.println(ex.getMessage());
-            } catch (StringIndexOutOfBoundsException ex) {
-                System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
-            }
-            catch (DateTimeParseException ex) {
-                System.out.println("Invalid date and time format\n"
-                        + "Please enter date and time as 'dd/MM/yyyy HHmm'");
+                String fullCommand = ui.readCommand();
```

Comment:
Your main method is very short and readable now as a result of adding command classes. Very nice and well done!